src/coreclr/jit/gcencode.cpp

```diff
@@ -1578,6 +1578,15 @@ size_t GCInfo::gcInfoBlockHdrSave(
         assert(header->epilogCount <= 1);
     }
 #endif
+    if (compiler->UsesFunclets() && compiler->info.compFlags & CORINFO_FLG_SYNCH && compiler->opts.compDbgEnC)
```

Comment:
```suggestion
    if (compiler->UsesFunclets() && compiler->info.compFlags & CORINFO_FLG_SYNCH)
```

We need this information even for non-EnC to get position of generic parameters.

Reply:
For reference, this is exposed with the Pri1 tests with `DOTNET_GCStress=0x3` in `JIT\Generics\Fields\getclassfrommethodparam\getclassfrommethodparam.dll`.

Reply:
I've addressed this issue as well as an issue where the method is both synchronized and has a stack allocation in it.